src/main/java/com/epam/esm/controller/TagController.java

```diff
@@ -0,0 +1,49 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.dao.DaoException;
+import com.epam.esm.model.Tag;
+import com.epam.esm.service.CertificateServiceException;
+import com.epam.esm.service.TagService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.MultiValueMap;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
```

Comment:
How many annotations did you use?

Reply:
@RestController @RequestMapping @RequestParam @RequestBody @Validated